src/shogun/neuralnets/NeuralNetwork.cpp

```diff
@@ -265,6 +266,10 @@ bool CNeuralNetwork::train_gradient_descent(SGMatrix<float64_t> inputs,
 		"Gradient descent learning rate (%f) must be > 0\n", gd_learning_rate);
 	REQUIRE(gd_momentum>=0,
 		"Gradient descent momentum (%f) must be > 0\n", gd_momentum);
+  REQUIRE(rms_epsilon>=0,
```

Comment:
Here and through the changes, new code's indentation doesn't match original one.